dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,331 @@
+			}
+		})
+	}
+}
+
+func Test_disjointBlockSet_addHeader(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		limit            int
+		blocks           map[common.Hash]*pendingBlock
+		parentToChildren map[common.Hash]map[common.Hash]struct{}
+		timeNow          func() time.Time
+	}
+	type args struct {
+		header *types.Header
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		err    error
+	}{
+		{
+			name:   "add header beyond capactiy",
+			fields: fields{},
+			args: args{header: &types.Header{
+				Number: 1,
+			}},
+			err: errors.New("cannot add block; set is at capacity"),
+		},
+		{
+			name: "add header",
+			fields: fields{
+				blocks:           make(map[common.Hash]*pendingBlock),
+				limit:            1,
+				timeNow:          time.Now,
+				parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
+			},
+			args: args{header: &types.Header{
+				Number: 1,
+			}},
+			err: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &disjointBlockSet{
+				limit:            tt.fields.limit,
+				blocks:           tt.fields.blocks,
+				parentToChildren: tt.fields.parentToChildren,
+				timeNow:          tt.fields.timeNow,
+			}
+			err := s.addHeader(tt.args.header)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_disjointBlockSet_clearBlocks(t *testing.T) {
+	t.Parallel()
+
+	testBlock := &pendingBlock{
+		clearAt: time.Now(),
+	}
+	type fields struct {
+		limit            int
+		blocks           map[common.Hash]*pendingBlock
+		parentToChildren map[common.Hash]map[common.Hash]struct{}
+		timeNow          func() time.Time
+	}
+	tests := []struct {
+		name      string
+		fields    fields
+		remaining int
+	}{
+		{
+			name: "base case",
+			fields: fields{
+				limit: 0,
+				blocks: map[common.Hash]*pendingBlock{
+					{}: testBlock,
+				},
+				timeNow: time.Now,
+			},
+			remaining: 0,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &disjointBlockSet{
+				limit:            tt.fields.limit,
+				blocks:           tt.fields.blocks,
+				parentToChildren: tt.fields.parentToChildren,
+				timeNow:          tt.fields.timeNow,
+			}
+			s.clearBlocks()
+			assert.Equal(t, tt.remaining, len(tt.fields.blocks))
+		})
+	}
+}
+
+func Test_disjointBlockSet_getBlocks(t *testing.T) {
+	t.Parallel()
+
+	testBlock := &pendingBlock{}
+	type fields struct {
+		blocks map[common.Hash]*pendingBlock
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   []*pendingBlock
+	}{
+		{
+			name:   "no blocks",
+			fields: fields{},
+			want:   []*pendingBlock{},
+		},
+		{
+			name: "base case",
+			fields: fields{
+				blocks: map[common.Hash]*pendingBlock{
+					{}: testBlock,
+				},
+			},
+			want: []*pendingBlock{testBlock},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
```suggestion
	for _, tt := range tests {
		tt := tt
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()
```

:warning: Also please add `tparallel` to the list of linters in `.golangci.yml`, that will warn you to run subtests in parallel for parent tests running in parallel :pray:

Reply:
Updated.

Reply:
Awesome!